Type confusion (?) triggered by interface composition

### The following program `sample.go` triggers an unexpected result

```go
package main

import (
	"fmt"

	"github.com/traefik/yaegi/interp"
	"github.com/traefik/yaegi/stdlib"
)

func main() {
	interpreter := interp.New(interp.Options{GoPath: "./gopath"})

	err := interpreter.Use(stdlib.Symbols)
	if err != nil {
		panic(err)
	}

	_, err = interpreter.Eval(`import "github.com/ip2location/ip2location-go"`)
	if err != nil {
		panic(err)
	}

	fmt.Println("no panic")
}
```


### Expected result

```console
no panic
```


### Got

```console
panic: 1:21: import "github.com/ip2location/ip2location-go" error: gopath/src/github.com/ip2location/ip2location-go/ip2location.go:246:23: cannot use type []uint8 as type int64
```


### Yaegi Version

d3bbe01d5c26

### Additional Notes

yaegi appears to confuse the expected type of the `data` argument in [line `246` of `ip2location.go`](https://github.com/ip2location/ip2location-go/blob/v9.1.0/ip2location.go#L246):

```go
func (d *DB) readuint8(pos int64) (uint8, error) {
	var retval uint8
	data := make([]byte, 1)
	_, err := d.f.ReadAt(data, pos-1)
	if err != nil {
		return 0, err
	}
	retval = data[0]
	return retval, nil
}
```

`d.f` is of type `DBReader`, which itself [is an interface](https://github.com/ip2location/ip2location-go/blob/v9.1.0/ip2location.go#L19) composed of `io.ReadCloser` and `io.ReaderAt`:

```go
type DBReader interface {
	io.ReadCloser
	io.ReaderAt
}
```

According to the [`io.ReaderAt` specification](https://pkg.go.dev/io#ReaderAt), `data` / `p` is of type `[]byte` (so `[]uint8`). But yaegi expects `int64` for some reason.

The implementation of `DBReader` at runtime [is always `os.File`](https://github.com/ip2location/ip2location-go/blob/v9.1.0/ip2location.go#L332). If all references to `DBReader` are replaced with `*os.File`, yaegi can import the package just fine, and it works as expected.

--- a/_test/issue-1280.go
+++ b/_test/issue-1280.go
@@ -1,0 +1,30 @@
+package main
+
+import (
+	"io"
+	"log"
+	"os"
+)
+
+type DBReader interface {
+	io.ReadCloser
+	io.ReaderAt
+}
+
+type DB struct {
+	f DBReader
+}
+
+func main() {
+	f, err := os.Open("/dev/null")
+	if err != nil {
+		log.Fatal(err)
+	}
+	d := &DB{f}
+	data := make([]byte, 1)
+	_, _ = d.f.ReadAt(data, 0)
+	println("bye")
+}
+
+// Output:
+// bye

--- a/interp/type.go
+++ b/interp/type.go
@@ -1071,7 +1071,7 @@ func (t *itype) in(i int) *itype {
 		return t.arg[i]
 	case valueT:
 		if t.rtype.Kind() == reflect.Func {
-			if t.recv != nil {
+			if t.recv != nil && !isInterface(t.recv) {
 				i++
 			}
 			if t.rtype.IsVariadic() && i == t.rtype.NumIn()-1 {

